src/sonic-bgpcfgd/bgpcfgd/managers_advertise_rt.py

```diff
@@ -18,82 +22,140 @@ def __init__(self, common_objs, db, table):
+        return False
 
-    def add_route_advertisement(self, vrf, ip_prefix):
+    def _del_handler_validate(self, key):
+        return self._ip_addr_validate(key)
+
+    def add_route_advertisement(self, vrf, ip_prefix, data):
         if self.directory.path_exist("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME, "localhost/bgp_asn"):
-            if not self.advertised_routes.get(vrf, set()):
+            if not self.advertised_routes.get(vrf, dict()):
                 self.bgp_network_import_check_commands(vrf, self.OP_ADD)
-            self.advertise_route_commands(ip_prefix, vrf, self.OP_ADD)
-
-        self.advertised_routes.setdefault(vrf, set()).add(ip_prefix)
+            self.advertise_route_commands(ip_prefix, vrf, self.OP_ADD, data)
 
+        self.advertised_routes.setdefault(vrf, dict()).update({ip_prefix: data})
 
     def remove_route_advertisement(self, vrf, ip_prefix):
-        self.advertised_routes.setdefault(vrf, set()).discard(ip_prefix)
-        if not self.advertised_routes.get(vrf, set()):
+        if ip_prefix not in self.advertised_routes.get(vrf, dict()):
+            log_info("BGPAdvertiseRouteMgr:: %s|%s does not exist" % (vrf, ip_prefix))
+            return
+        self.advertised_routes.get(vrf, dict()).pop(ip_prefix)
+        if not self.advertised_routes.get(vrf, dict()):
             self.advertised_routes.pop(vrf, None)
 
         if self.directory.path_exist("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME, "localhost/bgp_asn"):
-            if not self.advertised_routes.get(vrf, set()):
+            if not self.advertised_routes.get(vrf, dict()):
                 self.bgp_network_import_check_commands(vrf, self.OP_DELETE)
             self.advertise_route_commands(ip_prefix, vrf, self.OP_DELETE)
 
-
-    def advertise_route_commands(self, ip_prefix, vrf, op):
+    def advertise_route_commands(self, ip_prefix, vrf, op, data=None):
         is_ipv6 = TemplateFabric.is_ipv6(ip_prefix)
-        bgp_asn = self.directory.get_slot("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME)["localhost"]["bgp_asn"]
-
+        bgp_asn = self.directory.get_slot("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME)["localhost"][
+            "bgp_asn"
+        ]
         cmd_list = []
-        if vrf == 'default':
+        if vrf == "default":
             cmd_list.append("router bgp %s" % bgp_asn)
         else:
             cmd_list.append("router bgp %s vrf %s" % (bgp_asn, vrf))
 
         cmd_list.append(" address-family %s unicast" % ("ipv6" if is_ipv6 else "ipv4"))
-        cmd_list.append("  %snetwork %s" % ('no ' if op == self.OP_DELETE else '', ip_prefix))
 
-        self.cfg_mgr.push_list(cmd_list)
+        if data and "profile" in data:
+            cmd_list.append("  network %s route-map %s" % (ip_prefix, "%s_RM" % data["profile"]))
+            log_info(
+                "BGPAdvertiseRouteMgr:: Update bgp %s network %s with route-map %s"
```

Comment:
Based on scale, there could be significant number of route updates. So please reduce the log-level and ensure it won't show up in regular flow unless explicitly enabled

Reply:
Good one, change to debug.